lsp-types/src/Language/LSP/Types/Common.hs

```diff
@@ -10,14 +13,41 @@ module Language.LSP.Types.Common (
   , List (..)
   , Empty (..)
   , Int32
-  , Word32 ) where
+  , UInt ) where
 
 import Control.Applicative
 import Control.DeepSeq
 import Data.Aeson
+import Data.Coerce (coerce)
+import Data.Euclidean as E
 import Data.Int (Int32)
-import Data.Word (Word32)
-import GHC.Generics
+import Data.Mod.Word
+import Text.Read (Read(readPrec))
+import GHC.Generics hiding (UInt)
+
+-- | The "uinteger" type in the LSP spec.
+--
+-- Unusually, this is a **31**-bit unsigned integer, not a 32-bit one.
+newtype UInt = UInt (Mod 31)
```

Comment:
```suggestion
newtype UInt = UInt (Mod 2147483648)
```
Or `2 ^ 31`, if you import `GHC.TypeNats`.

Reply:
:facepalm: 

Fortunately, the HLS tests do fail with this, so I was going to notice eventually...